contracts/base/Settings.sol

```diff
@@ -363,6 +370,75 @@ contract Settings is Pausable, SettingsInterface {
         );
     }
 
+    /**
+        @notice Add a new Node Component with its version.
+        @param componentName name of the component to be added.
+        @param minVersion minimum component version supported.
+     */
+    function createComponentVersion(bytes32 componentName, uint32 minVersion)
+        external
+        onlyPauser()
+        whenNotPaused()
+    {
+        require(minVersion > 0, "INVALID_COMPONENT_VERSION");
+        require(componentName != "", "COMPONENT_NAME_MUST_BE_PROVIDED");
+        require(componentVersions[componentName] == 0, "COMPONENT_ALREADY_EXISTS");
+        componentVersions[componentName] = minVersion;
+        emit ComponentVersionCreated(msg.sender, componentName, minVersion);
+    }
+
+    /**
+        @notice Remove a Node Component from the list.
+        @param componentName name of the component to be removed.
+     */
+    function removeComponentVersion(bytes32 componentName)
+        external
+        onlyPauser()
+        whenNotPaused()
+    {
+        require(componentName != "", "COMPONENT_NAME_MUST_BE_PROVIDED");
+        require(componentVersions[componentName] > 0, "COMPONENT_NOT_FOUND");
+        uint32 previousVersion = componentVersions[componentName];
+        delete componentVersions[componentName];
+        emit ComponentVersionRemoved(msg.sender, componentName, previousVersion);
+    }
+
+    /**
+        @notice Get the version of a specific node component.
+        @param componentName name of the component to return the version.
+        @return version of the node component if exists or zero 0 if not found.
+     */
+    function getComponentVersion(bytes32 componentName) external view returns (uint32) {
+        return componentVersions[componentName];
+    }
+
+    /**
+        @notice Set a new version for a Node Component.
+        @param componentName name of the component to be modified.
+        @param newVersion minimum component version supported.
+     */
+    function updateComponentVersion(bytes32 componentName, uint32 newVersion)
+        external
+        onlyPauser()
+        whenNotPaused()
+    {
+        require(newVersion > 0, "INVALID_COMPONENT_VERSION");
```

Comment:
I think as we are validating "componentVersions[componentName] > 0" and "newVersion > componentVersions[componentName]", then the validation the "newVersion > 0' validation is not required.

Reply:
Redundant validation has been removed